### ISSUE TITLE
Support AWS AppFlow

### ISSUE THREAD

**Issue by jasonstitt** [hashicorp/terraform-provider-aws#16253]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS AppFlow allows setting up integration points with various SaaS tools and AWS locations such as S3. This would be great to have in Terraform because of the repetitive and boilerplate nature of some of these setups - e.g. Salesforce connector setup is on a per-object (type) basis so there could be a ton of these.

### New or Affected Resource(s)

* aws_appflow_flow
* aws_appflow_connector_profile

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

https://aws.amazon.com/blogs/aws/new-announcing-amazon-appflow/

https://docs.aws.amazon.com/appflow/1.0/APIReference/Welcome.html

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appflow-flow.html

**Comment by jverhoeks** [comment]
Is there any progress on this?

**Comment by mrampf** [comment]
Any updates on this? Do you need any help for this since we have various customers asking for this support.

**Comment by ClemensSelbach** [comment]
are there any news on this topic... we need this asap!

**Comment by udaykiran19** [comment]
Please help on the planned timeline/roadmap for the appflow release ?
We are waiting on this for a long time now

**Comment by ksharlandjiev** [comment]
Hi all. 
I'm an AppFlow SME.
I speak with lots of customers daily, who wants to see amazon AppFlow support . 

Is there anything I can help to speed up the process? 
Please get in touch.

**Comment by breathingdust** [comment]
Hi all 👋  Just letting you know that this is issue is featured on this quarters [roadmap](https://github.com/hashicorp/terraform-provider-aws/blob/main/ROADMAP.md). If a PR exists to close the issue a maintainer will review and either make changes directly, or work with the original author to get the contribution merged. If you have written a PR to resolve the issue please ensure the "Allow edits from maintainers" box is checked. Thanks for your patience and we are looking forward to getting this merged soon!

**Comment by szemek** [comment]
Hi @breathingdust @zhelding
I'm trying to refresh a code from my old pull request https://github.com/hashicorp/terraform-provider-aws/pull/19543
New pull request is here https://github.com/hashicorp/terraform-provider-aws/pull/23892, but acceptance tests stopped working properly. Could you take a look and give me some hints? I appreciate any guidance on that.
Something is wrong with an implementation of `GetConnectorProfile` used by `resourceConnectorProfileRead`.

**Comment by ksharlandjiev** [comment]
Hey any update on this? Thanks

**Comment by AdamTylerLynch** [comment]
Nice job @zhelding ! 👏👏👏

**Comment by zhelding** [comment]
Hi folks. With #24017 adding support for `appflow_flow` resources and #23892 adding support for `appflow_connector_profile` resources, we now have basic support for Amazon AppFlow! 

Huge thanks to @szemek for his contributions. 

Rendered docs for both [`appflow_flow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appflow_flow) and [`appflow_connector_profile`](https://github.com/hashicorp/terraform-provider-aws/pull/23892) are available on the Terraform Registry.

If any issues arise when using these resources, or if any additional AppFlow coverage / functionality is needed -- please open an additional issue describing the desired behavior.

**Comment by dalareo** [comment]
There is a bug in flow.go file:

```
"task_properties": {
							Type:         schema.TypeMap,
							Optional:     true,
							ValidateFunc: validation.StringInSlice(appflow.OperatorPropertiesKeys_Values(), false),
							Elem: &schema.Schema{
								Type:         schema.TypeString,
								ValidateFunc: validation.All(validation.StringMatch(regexp.MustCompile(`\S+`), "must not contain any whitespace characters"), validation.StringLenBetween(0, 2048)),
							},
```

Map is validated against a string (OperatorPropertiesKeys_values()) so neither a map of strings or a string is accepted here. We are working in a PR for this.

**Comment by zhelding** [comment]
Hi @dalareo: nice catch! Tracking bug with issue #24888.
